File: Android_Zone_Test/src/com/example/mylib_test/activity/animal/XfermodeActivity.java
package com.example.mylib_test.activity.animal;

import android.view.View;
import android.widget.Button;

import com.example.mylib_test.R;
import com.example.mylib_test.activity.animal.helper.PopXfermode;
import com.example.mylib_test.activity.animal.viewa.XfermodeView;

import com.zone.lib.base.activity.BaseActivity;


/**
 * Created by Administrator on 2016/3/21.
 */
public class XfermodeActivity extends BaseActivity {
    private View rl_main;
    private Button bt_pop;
    private PopXfermode popXfermode;
    private XfermodeView xfermodeView;
    private Button bt_borderIsCircle;

    @Override
    public void setContentView() {
        setContentView(R.layout.a_animal_xfermode);
    }

    @Override
    public void findIDs() {
        rl_main = findViewById(R.id.rl_main);
        bt_pop = (Button) findViewById(R.id.bt_pop);
        bt_borderIsCircle = (Button) findViewById(R.id.borderIsCircle);
        xfermodeView = (XfermodeView) findViewById(R.id.xfermodeView);
        xfermodeView.setBordMode(borderIsCircle);
    }

    @Override
    public void initData() {
        popXfermode = new PopXfermode(this, xfermodeView, bt_pop);
    }

    @Override
    public void setListener() {

    }

    boolean borderIsCircle = false;

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.bt_pop:
                popXfermode.show();
                break;
            case R.id.borderIsCircle:
                borderIsCircle=!borderIsCircle;
                xfermodeView.setBordMode(borderIsCircle);
                if (borderIsCircle)
                    bt_borderIsCircle.setText("边界是圆");
                else
                    bt_borderIsCircle.setText("边界是手机大小");
                break;
        }
        super.onClick(v);
    }

}

File: Android_Zone_Test/src/com/example/mylib_test/activity/animal/XfermodeActivity.kt
package com.example.mylib_test.activity.animal

import android.view.View

import com.example.mylib_test.R
import com.example.mylib_test.activity.animal.helper.PopXfermode
import com.zone.lib.base.controller.activity.BaseFeatureActivity

import kotlinx.android.synthetic.main.a_animal_xfermode.*


/**
 * Created by Administrator on 2016/3/21.
 */
class XfermodeActivity : BaseFeatureActivity() {
    private lateinit var popXfermode: PopXfermode

    internal var borderIsCircleProperty = false

    override fun setContentView() {
        setContentView(R.layout.a_animal_xfermode)
    }

    override fun initData() {
        xfermodeView.setBordMode(borderIsCircleProperty)
        popXfermode = PopXfermode(this, xfermodeView, bt_pop)
    }

    override fun setListener() {

    }

    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.bt_pop -> popXfermode.show()
            R.id.borderIsCircle -> {
                borderIsCircleProperty = !borderIsCircleProperty
                xfermodeView.setBordMode(borderIsCircleProperty)
                if (borderIsCircleProperty)
                    borderIsCircle.text = "边界是圆"
                else
                    borderIsCircle.text = "边界是手机大小"
            }
            else -> {
            }
        }
        super.onClick(v)
    }
}
